packages/rx_bloc_cli/mason_templates/bricks/lib_auth_matrix/__brick__/lib/lib_auth_matrix/models/auth_matrix_response.dart

```diff
@@ -1,30 +1,94 @@
 {{> licence.dart }}
 
+import 'package:copy_with_extension/copy_with_extension.dart';
+import 'package:equatable/equatable.dart';
 import 'package:json_annotation/json_annotation.dart';
 
-import 'auth_matrix_action_type.dart';
+import 'auth_matrix_method.dart';
+
+import 'payload/auth_matrix_payload_response.dart';
+import 'payload/response/auth_matrix_last_login_payload_response.dart';
 
 part 'auth_matrix_response.g.dart';
 
 @JsonSerializable()
-class AuthMatrixResponse {
-  AuthMatrixResponse(this.transactionId, this.authZ, this.status);
+@CopyWith()
+class AuthMatrixResponse with EquatableMixin {
+  AuthMatrixResponse({
+    required this.securityToken,
+    required this.authMethod,
+    required this.transactionId,
+    required this.documentIds,
+    required this.expires,
+    this.payload,
+  });
+
+  /// The draft document ids created by the initial auth matrix request.
+  ///
+  /// Once the user completes all authentication methods successfully, the documents will be marked as signed.
+  final List<int> documentIds;
+
+  /// The security token, which ensures the chain of auth matrix requests.
+  ///
+  /// It's used in the following payload requests to ensure the chain of requests.
+  @JsonKey(defaultValue: '')
+  final String securityToken;
 
+  /// The expiration date of the auth matrix request.
+  @JsonKey(defaultValue: '')
+  final String expires;
+
+  /// The auth matrix method to be used for the following request.
+  final AuthMatrixMethod authMethod;
+
+  /// The unique auth matrix transaction id.
+  ///
+  /// It's used in the following payload requests to ensure the chain of requests.
   final String transactionId;
-  @JsonKey(unknownEnumValue: AuthMatrixActionType.none)
-  final AuthMatrixActionType authZ;
-  @JsonKey(unknownEnumValue: AuthMatrixStatus.unknown)
-  final AuthMatrixStatus status;
+
+  /// Dynamic additional data to be received from the API along with the required
+  /// response properties. Create custom AuthMatrixResponsePayload implementation
```

Comment:
Here it says `AuthMatrixResponsePayload` in the documentation, however the class is called `AuthMatrixPayloadResponse`